[Bug]: @storybook/react-webpack5 v7.3 crashes without @storybook/manager-api

### Describe the bug

When updating `@storybook/react-webpack5` to v7.3, Storybook failed to launch. After debugging, I found Storybook was not able to resolve `@storybook/manager-api`. Before v7.3, `@storybook/react-webpack5` shipped with `@storybook/manager-api` in `dependencies`. 

The fix is simple: consumers need to install `@storybook/manager-api`, however this should either be shipped within `@storybook/react-webpack5` or listed as a peer dependency

Crash log
```
> storybook build --webpack-stats-json -o storybook-static
@storybook/cli v7.3.1
info => Cleaning outputDir: /storybook-static
Attention: We've detected that you're using the following addons in versions which are known to be incompatible with Storybook 7:
- @storybook/addon-queryparams@6.2.9
Please be aware they might not work in Storybook 7. Reach out to their maintainers for updates and check the following Github issue for more information:
https://github.com/storybookjs/storybook/issues/20529
info => Loading presets
info Found existing addon "@storybook/addon-actions", skipping.
info Found existing addon "@storybook/addon-toolbars", skipping.
WARN   Failed to load preset: "/builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js"
ERR! Error: Cannot find module '@storybook/manager-api/package.json'
ERR! Require stack:
ERR! - /builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js
ERR! - /builds/node_modules/@storybook/core-common/dist/index.js
ERR! - /builds/node_modules/@storybook/telemetry/dist/index.js
ERR! - /builds/node_modules/@storybook/cli/dist/generate.js
ERR! - /builds/node_modules/@storybook/cli/bin/index.js
ERR! - /builds/node_modules/storybook/index.js
ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)
ERR!     at Function.Module._resolveFilename (/builds/node_modules/esbuild-register/dist/node.js:4768:36)
ERR!     at Function.resolve (node:internal/modules/cjs/helpers:125:19)
ERR!     at getAbsolutePath (/builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js:3:323)
ERR!     at Object.<anonymous> (/builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js:3:785)
ERR!     at Module._compile (node:internal/modules/cjs/loader:1198:14)
ERR!     at Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
ERR!     at Object.newLoader (/builds/node_modules/esbuild-register/dist/node.js:2262:9)
ERR!     at Object.extensions..js (/builds/node_modules/esbuild-register/dist/node.js:4807:24)
ERR!     at Module.load (node:internal/modules/cjs/loader:1076:32)
ERR!     at Function.Module._load (node:internal/modules/cjs/loader:911:12)
ERR!     at Module.require (node:internal/modules/cjs/loader:1100:19)
ERR!     at require (node:internal/modules/cjs/helpers:119:18)
ERR!     at interopRequireDefault (/builds/node_modules/@storybook/core-common/dist/index.js:6:21)
ERR!     at getContent (/builds/node_modules/@storybook/core-common/dist/index.js:13:332)
ERR!     at loadPreset (/builds/node_modules/@storybook/core-common/dist/index.js:13:517)
ERR!  Error: Cannot find module '@storybook/manager-api/package.json'
ERR! Require stack:
ERR! - /builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js
ERR! - /builds/node_modules/@storybook/core-common/dist/index.js
ERR! - /builds/node_modules/@storybook/telemetry/dist/index.js
ERR! - /builds/node_modules/@storybook/cli/dist/generate.js
ERR! - /builds/node_modules/@storybook/cli/bin/index.js
ERR! - /builds/node_modules/storybook/index.js
ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1028:15)
ERR!     at Function.Module._resolveFilename (/builds/node_modules/esbuild-register/dist/node.js:4768:36)
ERR!     at Function.resolve (node:internal/modules/cjs/helpers:125:19)
ERR!     at getAbsolutePath (/builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js:3:323)
ERR!     at Object.<anonymous> (/builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js:3:785)
ERR!     at Module._compile (node:internal/modules/cjs/loader:1198:14)
ERR!     at Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
ERR!     at Object.newLoader (/builds/node_modules/esbuild-register/dist/node.js:2262:9)
ERR!     at Object.extensions..js (/builds/node_modules/esbuild-register/dist/node.js:4807:24)
ERR!     at Module.load (node:internal/modules/cjs/loader:1076:32)
ERR!     at Function.Module._load (node:internal/modules/cjs/loader:911:12)
ERR!     at Module.require (node:internal/modules/cjs/loader:1100:19)
ERR!     at require (node:internal/modules/cjs/helpers:119:18)
ERR!     at interopRequireDefault (/builds/node_modules/@storybook/core-common/dist/index.js:6:21)
ERR!     at getContent (/builds/node_modules/@storybook/core-common/dist/index.js:13:332)
ERR!     at loadPreset (/builds/node_modules/@storybook/core-common/dist/index.js:13:517) {
ERR!   code: 'MODULE_NOT_FOUND',
ERR!   requireStack: [
ERR!     '/builds/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js',
ERR!     '/builds/node_modules/@storybook/core-common/dist/index.js',
ERR!     '/builds/node_modules/@storybook/telemetry/dist/index.js',
ERR!     '/builds/node_modules/@storybook/cli/dist/generate.js',
ERR!     '/builds/node_modules/@storybook/cli/bin/index.js',
ERR!     '/builds/node_modules/storybook/index.js'
ERR!   ]
ERR! }
```

### To Reproduce

Install Storybook with `@storybook/react-webpack5` preset but do not install `@storybook/manager-api`. Attempt to launch Storybook

### System

```shell
Environment Info:

  System:
    OS: macOS 13.5
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  Binaries:
    Node: 16.20.2 - ~/.nvm/versions/node/v16.20.2/bin/node
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.2/bin/npm
  Browsers:
    Chrome: 115.0.5790.170
    Firefox: 115.0.3
    Safari: 16.6
  npmPackages:
    @storybook/addon-a11y: ^7.3.1 => 7.3.1 
    @storybook/addon-actions: ^7.3.1 => 7.3.1 
    @storybook/addon-essentials: ^7.3.1 => 7.3.1 
    @storybook/addon-interactions: ^7.3.1 => 7.3.1 
    @storybook/addon-links: ^7.3.1 => 7.3.1 
    @storybook/addon-queryparams: ^6.2.9 => 6.2.9 
    @storybook/addon-toolbars: ^7.3.1 => 7.3.1 
    @storybook/jest: ^0.1.0 => 0.1.0 
    @storybook/manager-api: ^7.3.1 => 7.3.1 
    @storybook/react: ^7.3.1 => 7.3.1 
    @storybook/react-webpack5: ^7.3.1 => 7.3.1 
    @storybook/test-runner: ^0.13.0 => 0.13.0 
    @storybook/testing-library: ^0.2.0 => 0.2.0
```


### Additional context

New bug as of version 7.3.0. Most likely from [commit f70f0508f182b475c65a5b9d87444421ef8790eb ](https://github.com/storybookjs/storybook/pull/23496/commits/f70f0508f182b475c65a5b9d87444421ef8790eb)

--- a/code/builders/builder-webpack5/package.json
+++ b/code/builders/builder-webpack5/package.json
@@ -56,15 +56,23 @@
     "prep": "../../../scripts/prepare/bundle.ts"
   },
   "dependencies": {
-    "@babel/core": "^7.22.0",
+    "@babel/core": "^7.22.9",
+    "@storybook/addons": "workspace:*",
     "@storybook/channels": "workspace:*",
+    "@storybook/client-api": "workspace:*",
     "@storybook/client-logger": "workspace:*",
+    "@storybook/components": "workspace:*",
     "@storybook/core-common": "workspace:*",
     "@storybook/core-events": "workspace:*",
     "@storybook/core-webpack": "workspace:*",
+    "@storybook/global": "^5.0.0",
+    "@storybook/manager-api": "workspace:*",
     "@storybook/node-logger": "workspace:*",
     "@storybook/preview": "workspace:*",
     "@storybook/preview-api": "workspace:*",
+    "@storybook/router": "workspace:*",
+    "@storybook/store": "workspace:*",
+    "@storybook/theming": "workspace:*",
     "@swc/core": "^1.3.49",
     "@types/node": "^16.0.0",
     "@types/semver": "^7.3.4",
@@ -101,6 +109,10 @@
     "pretty-hrtime": "^1.0.3",
     "slash": "^5.0.0",
     "typescript": "~4.9.3"
+  },
+  "peerDependencies": {
+    "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
+    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0"
   },
   "peerDependenciesMeta": {
     "typescript": {

--- a/code/builders/builder-webpack5/src/preview/iframe-webpack.config.ts
+++ b/code/builders/builder-webpack5/src/preview/iframe-webpack.config.ts
@@ -36,29 +36,19 @@ const storybookPaths: Record<string, string> = {
     `@storybook/components`
   )}/dist/experimental`,
   ...[
-    // these packages are not pre-bundled because of react dependencies.
-    // these are not dependencies of the builder anymore, thus resolving them can fail.
-    // we should remove the aliases in 8.0, I'm not sure why they are here in the first place.
+    // these packages are not pre-bundled because of react dependencies
     'components',
     'global',
     'manager-api',
     'router',
     'theming',
-  ].reduce((acc, sbPackage) => {
-    let packagePath;
-    try {
-      packagePath = getAbsolutePath(`@storybook/${sbPackage}`);
-    } catch (e) {
-      // ignore
-    }
-    if (packagePath) {
-      return {
-        ...acc,
-        [`@storybook/${sbPackage}`]: getAbsolutePath(`@storybook/${sbPackage}`),
-      };
-    }
-    return acc;
-  }, {}),
+  ].reduce(
+    (acc, sbPackage) => ({
+      ...acc,
+      [`@storybook/${sbPackage}`]: getAbsolutePath(`@storybook/${sbPackage}`),
+    }),
+    {}
+  ),
   // deprecated, remove in 8.0
   [`@storybook/api`]: getAbsolutePath(`@storybook/manager-api`),
 };

